charms/kfp-profile-controller/tests/integration/test_charm.py

```diff
@@ -163,6 +166,31 @@ def validate_profile_resources(
     assert expected_label_value == namespace.metadata.labels[expected_label]
 
 
+@retry(
+    wait=wait_exponential(multiplier=1, min=1, max=10),
+    stop=stop_after_delay(30),
+    reraise=True,
+)
+def validate_profile_deployments_with_custom_images(
+    lightkube_client: lightkube.Client,
+    profile_name: str,
+):
+    # Get deployments
+    pipeline_ui_deployment = lightkube_client.get(
+        Deployment, name="ml-pipeline-ui-artifact", namespace=profile_name
+    )
+    visualization_server_deployment = lightkube_client.get(
+        Deployment, name="ml-pipeline-visualizationserver", namespace=profile_name
+    )
+
+    # Assert images
+    assert pipeline_ui_deployment.spec.template.spec.containers[0].image == CUSTOM_FRONTEND_IMAGE
+    assert (
+        visualization_server_deployment.spec.template.spec.containers[0].image
+        == CUSTOM_VISUALISATION_IMAGE
+    )
```

Comment:
Could we make `frontend_image` and `visualization_image` arguments to this helper?  It feels a bit hidden that the helper uses globals.  `test_new_images` can still call the function with the globals, but it'll be a bit clearer that way